contrib/devtools/update-rust-hashes.py

```
@@ -0,0 +1,125 @@
+
+def get_rust_version(makefile_path: Path) -> str:
+    content = makefile_path.read_text()
+    match = re.search(r"\$\(package\)_version:=(.+)", content)
+    if not match:
+        raise RuntimeError("Could not find Rust version in makefile")
+    return match.group(1).strip()
+
+
+def compute_sha256(url: str) -> str:
+    hasher = hashlib.sha256()
+    with urllib.request.urlopen(url) as response:
```

Ruff (0.16.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)